src/NexusMods.Paths/AbsolutePath.cs

```diff
@@ -109,6 +125,75 @@ public static AbsolutePath FromFullPath(string fullPath, IFileSystem? fileSystem
     public static AbsolutePath FromDirectoryAndFileName(string? directoryPath, string fullPath, IFileSystem? fileSystem = null)
         => new(directoryPath, fullPath, fileSystem ?? Paths.FileSystem.Shared);
 
+    /// <summary>
+    /// Returns true if the given directory is a root directory on the current
+    /// platform.
+    /// </summary>
+    /// <param name="directory"></param>
+    /// <returns></returns>
+    /// <exception cref="PlatformNotSupportedException">
+    /// The current platform is not supported.
+    /// </exception>
+    internal static bool IsRootDirectory(ReadOnlySpan<char> directory)
+    {
+        var rootLength = GetRootLength(directory);
+        return rootLength == directory.Length;
+    }
+
+    /// <summary>
+    /// Gets the length of the root of the path.
+    /// </summary>
+    /// <param name="path"></param>
+    /// <returns></returns>
+    /// <exception cref="PlatformNotSupportedException">
+    /// The current platform is not supported.
+    /// </exception>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static int GetRootLength(ReadOnlySpan<char> path)
+    {
+        if (OperatingSystem.IsLinux())
+            return path.Length >= 1 && path[0] == PathSeparatorForInternalOperations ? 1 : 0;
+
+        if (!OperatingSystem.IsWindows())
+            throw new PlatformNotSupportedException();
+
+        // NOTE (erri120): UNC paths (\\?\C:\) and Device paths (\\?\.)
+        // are not supported. Only classic drive specific paths: C:\
+
+        // https://github.com/dotnet/runtime/blob/bb2b8605df0e916dcd6339f2056efb2bd4521ff5/src/libraries/Common/src/System/IO/PathInternal.Windows.cs#L223-L233
+        if (path.Length < 3 || path[1] != ':' || !IsValidDriveChar(path[0])) return 0;
+        return path[2] == PathSeparatorForInternalOperations ? 3 : 0;
+    }
+
+    /// <summary>
+    /// Returns true if the given character is a valid drive letter.
+    /// </summary>
+    /// <param name="value"></param>
+    /// <returns></returns>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private static bool IsValidDriveChar(char value)
+    {
+        // Licensed to the .NET Foundation under one or more agreements.
+        // The .NET Foundation licenses this file to you under the MIT license.
+        // https://github.com/dotnet/runtime/blob/main/LICENSE.TXT
+        // source: https://github.com/dotnet/runtime/blob/d9f453924f7c3cca9f02d920a57e1477293f216e/src/libraries/Common/src/System/IO/PathInternal.Windows.cs#L69-L75
+        return (uint)((value | 0x20) - 'a') <= 'z' - 'a';
+    }
+
+    /// <summary>
+    /// Joins two path components together.
+    /// </summary>
+    /// <param name="left"></param>
+    /// <param name="right"></param>
+    /// <returns></returns>
+    private static string JoinPathComponents(string left, string right)
```

Comment:
Please use span [at least for `right`] and aggressive inline this.

Previous string joins in this PR were replaced with this method; and all spans were `ToString()'d` when we have no use for the new string, leading to allocations.

Normally not a big deal but this also affects `CombineUnchecked` and `GetFullPath` which I consider high performance APIs.